File: core/index/semantic_index_engine.py
from __future__ import annotations

from collections import defaultdict

from typing import Any, Dict, List


class SemanticIndex:

    def __init__(self) -> None:

        self._index: Dict[
            str,
            List[Dict[str, Any]]
        ] = defaultdict(list)

    def add(
        self,
        key: str,
        value: Dict[str, Any],
    ) -> None:

        self._index[key].append(value)

    def search(
        self,
        key: str,
    ) -> List[Dict[str, Any]]:

        return list(
            self._index.get(key, [])
        )

File: src/index/semanticIndexEngine.ts
/**
 * Converted from Python: core/index/semantic_index_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export class SemanticIndex {
  declare _index: any;
  constructor() {
    this._index = py.defaultdict(() => []);
  }
  add(key: any, value: any): any {
    py.listAppend(py.at(this._index, key), value);
  }
  search(key: any): any {
    return [...py.iter(py.get(this._index, key, []))];
  }
}
